### DISCUSSION TITLE
should the  cache names be made configurable?

### DISCUSSION THREAD

**Discussion by kkcmadhu** [keycloak/keycloak#15007] [Archive]
Key cloak presently uses well defined cache names linke realms, clients, users, users-sessions, client-scopes etc. while this is ok, but when using external cache infinispan server/hotrod, i might want to have a single infinispan cluster to server multiple deployments of keycloak,  instead of having one infispan cluster per keycloak deployment. if the cache name can be made configureable or if there is a possiblity to add a prefix to all cache names through -D arg, it might  be very very useful.

**Comment by sschu** [comment]
I am not an Infinispan expert, but couldn't you use different cache-containers with different names and have Keycloak caches with identical names inside these cache containers?

**Reply by kkcmadhu** [reply]
the cache container name too is fixed "keycloak" i guess. ya, if there are  ways to override cache container names that too should work..
also, in the hot rod urls, you can specify only  the infinispan cluster names, i am not sure if the hot rod uri can take cache container name..  or at the lease keycloak do not seem to be supporting it , please correct me if i am missing anything here..
especially when the keycloak is started like this 

```
bin/kc.sh start-dev --storage=jpa \
  --db-url=<jdbc-url> --db-username=<username> --db-password=<password> \
  --storage-hotrod-host=<host> --storage-hotrod-port=<port> \
  --storage-hotrod-username=<username> --storage-hotrod-password=<password> \
  --storage-area-action-token=hotrod \
  --storage-area-auth-session=hotrod \
  --storage-area-single-use-object=hotrod \
  --storage-area-user-session=hotrod
```

in this scenario the cluster name is assumed as "keycloak"

**Reply by kkcmadhu** [reply]
support for --storage-hotrod-container-name  in kc.sh should be sufficient  i guess.

**Reply by kkcmadhu** [reply]
i went through some infinispan docs, the doc never refer containers.. it just refers to cache names and  cache-manager ( the equivalent of container ) i guess..

recommendation is  to use differnt cache names for multitenancy / use cases where the protocol etc are same..
and use cache manager when  protocol etc are different
refer 
https://infinispan.org/docs/stable/titles/overview/overview.html#cache_manager_questions

**Comment by mhajas** [comment]
Thank you for creating the discussion @kkcmadhu! I created an enhancement issue for this: #17209.

Note: Can't say for sure, but we probably won't be able to work on this soon (KC23 or later).
